crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,143 +1,328 @@
+    /// 4. Aliases such as `@/`, `#`, `~`, or `%`
+    /// 5. Absolute and relative paths such as
     ///
-    /// ```js
-    /// import sibling from "./sibling";       // Import group 1
-    /// import { internal } from "#internal";  // Import group 1
-    /// import "z";  // Import group 2
-    /// import "a";  // Import group 3
-    /// import React from "react";         // Import group 4
-    /// import assert from "node:assert";  // Import group 4
+    /// Two imports/exports with the same category are sorted using a [natural sort order](https://en.wikipedia.org/wiki/Natural_sort_order) adapted to URLs, packages, and paths.
+    /// Notably, the order ensures that `a < A < b < B`.
+    /// The order takes also numbers into account, e.g. `a9 < a10`.
+    /// Let's just takes the following example to demonstrate this order.
+    ///
+    /// ```js,expect_diagnostic
+    /// import path from "node:path";
+    /// import example from "https://example.org";
+    /// import B from "./b.js";
+    /// import A from "../a.js";
+    /// import alias from "@/alias";
+    /// import myLib from "@my/lib";
+    /// import lib from "lib";
+    /// ```
+    ///
+    /// ## Last resort sorting
+    ///
+    /// If two imports/exports share the same source and are in the same chunk, then they are ordered according to their kind as follows:
+    ///
+    /// - Default type import
+    /// - Default import
+    /// - Combined default and namespace import
+    /// - Combined default and named import
+    /// - Namespace type import / Namespace type export
+    /// - Namespace import / Namespace export
+    /// - Named type import / Named type export
+    /// - Named import / Named export
+    ///
+    /// Also, import and exports with attributes are always placed first.
+    ///
+    /// ```js,expect_diagnostic
+    /// import * as namespaceImport from "same-source";
+    /// import type * as namespaceTypeImport from "same-source";
+    /// import { namedImport } from "same-source";
+    /// import type { namedTypeImport } from "same-source";
+    /// import defaultNamespaceCombined, * as namespaceCombined from "same-source";
+    /// import defaultNamedCombined, { namedCombined } from "same-source";
+    /// import defaultImport from "same-source";
+    /// import type defaultTypeImport from "same-source";
+    /// import { importWithAttribute } from "same-source" with { "attribute": "value" } ;
+    /// ```
+    ///
+    /// ## Comment handling
+    ///
+    /// When sorting imports and exports, attached comments are moved with their import or export,
+    /// and detached comments (comments followed by a blank line) are left where they are.
+    ///
+    /// Note that there is an exception at the rule.
+    /// If a comment appears at the top of the file, it is considered as detached even if no blank lien follows.
+    /// This ensures that copyright notice and file header comment stay at the top of the file.
```

Comment:
```suggestion
    /// However, there is an exception to the rule.
    /// If a comment appears at the top of the file, it is considered detached even if no blank line follows.
    /// This ensures that copyright notice and file header comments stay at the top of the file.
```

Be careful!! The first example that talked about attached comments had a comment at the top of the file, which is considered an attached comment 😄